parsec/core/logged_core.py

```diff
@@ -90,15 +91,15 @@ def _get_prevent_sync_pattern(prevent_sync_pattern_path: Path) -> Optional[Patte
         for line in data.splitlines():
             line = line.strip()
             if line and not line.startswith("#"):
-                regex.append(fnmatch.translate(line))
+                regex.append(str(Regex.from_pattern(line)))
         regex = "|".join(regex)
     except ValueError as exc:
         logger.warning(
             "Could not parse the file containing the filename patterns to ignore", exc_info=exc
         )
         return None
     try:
-        return re.compile(regex)
+        return Regex.from_regex_str(regex)
```

Comment:
1. Regex are compiled 2 time, could this be merged ?
2. We could use [`RegexSet`](https://docs.rs/regex/1.6.0/regex/struct.RegexSet.html)

Reply:
I don't know if this will change anything. A regex will be built when converting the `pattern` to an actual regex by the `glob_to_regex` function.
`RegexSet` expects an iterable of string-like object which means we will build `n` regexes and then an other bigger one to make the `RegexSet` object. Do you think `RegexSet` will be faster anyway?

Reply:
I took a more in depth view of `RegexSet` and as you said, we cannot build the set from a list of already compiled regexes but from string.
For me using `RegexSet` would convey the idea that is *Check if a string match those regexes* better.
But I do not find it pretty to build the `RegexSet` from a set of already compiled regexes that are converted to string.
As for speed I don't know which would be better.